boards: arduino_nano_33_ble: improve board

gmarull opened the pull request:
The board had numerous issues that had to be addressed (I wonder how a board with such poor quality got merged). Refer to each individual commit for details.

gmarull committed:
boards: arduino_nano_33_ble: remove redundant rtc1 entry

The peripheral is already disabled by default, so the entry was
pointless. Also removed commented code in init_sensors.c related to rtc.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: arduino_nano_33_ble: remove redundant comments from DT

Remove some comments that are redundant and provide no relevant
information.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: arduino_nano_33_ble: drop BLE_EN_USB_CONSOLE

Drop the BOARD_ARDUINO_NANO_33_BLE_EN_USB_CONSOLE Kconfig option. If
applications need a USB console, they need to adjust `prj.conf` (or
overlays) enabling relevant settings.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: arduino_nano_33_ble: remove redundant pwm entries

Enabling PWMs without selecting any channel is pointless, so remove
entries from DT. Also remove custom code in sensor_init, PWM API needs
to be used instead.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: arduino_nano_33_ble: remove undocumented debug code

It's not clear why the board configures debug/tracing functionality
using HAL low level code, remove it.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: arduino_nano_33_ble: define LED in DT

The board has a user LED attached to P1.09. This information was
hardcoded in board C files and enabled by default. This is something
applications need to do if needed (note that an always ON LED can incur
a considerable power consumption).

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: arduino_nano_33_ble: use regulator for sensors

Most sensors are enabled by enabling the VDD_ENV output. Define a
regulator on DT for that and enable the regulators if sensors are
enabled.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: arduino_nano_33_ble: refactor init code

- Specify in DT the pin to enable I2C pull-up connection
- Always build board.c, so that the I2C bus is always pulled-up

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: arduino_nano_33_ble: remove redundant "gpio library"

The board had an internal "library" to configure pins in an Arduinoeish
fasion. This is no longer used, so can be removed.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: arduino_nano_33_ble: remove redundant defconfigs

The I2C/SPI settings are automatically selected depending on DT
selection.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: arduino_nano_33_ble: improve DT formatting

Remove C99 like comments, add some extra spaces for better readability.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

MaureenHelm commented on boards/arm/arduino_nano_33_ble/arduino_nano_33_ble.dts:
```diff
@@ -35,13 +35,18 @@
 			label = "Blue LED";
 		};
+		led4: led_4 {
+			gpios = <&gpio1 9 0>;
```
Use `GPIO_ACTIVE_HIGH` or `GPIO_ACTIVE_LOW`

gmarull replied:
thanks, added `GPIO_ACTIVE_HIGH`, also adjusted other LEDs (`GPIO_ACTIVE_LOW`).

gmarull committed:
boards: arduino_nano_33_ble: add GPIO_ACTIVE_LOW to led0-2

The led0,1,2 LEDs are active low (MCU pins are electrically connected to
the LED cathodes).

Ref.
https://content.arduino.cc/assets/NANO33BLE_V4.0_sch.pdf

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull commented:
@carlescufi @MaureenHelm please re-approve :pray:

gmarull commented:
> @carlescufi @MaureenHelm please re-approve :pray:

@carlescufi @MaureenHelm please re-approve :pray: (rebased due to conflicts)

jfischer-no reviewed (approved):
Amen

rzr commented:
Could this change cause this observed regression:

https://github.com/zephyrproject-rtos/zephyr/issues/49480
